### PR TITLE
Hotfix rounding fields values on WBT write

### LINKED ISSUE #582: Rounding issues saving float values to WBT file
There are rounding issues writing float values to world file.
This only occurs for some specific values, for example 1.6 but not 1.5.

**Steps to Reproduce**
1. Open new world
2. Change `Viewpoint.far` to 1.6
3. Save the world
-> in the WBT file the far value is exported as `far 1.5999999999999998`

**Expected behavior**
The export field value should be `far 1.6` and not `far 1.5999999999999998`

**System**
 - Operating System: Linux Ubuntu 18.04

### PR DESCRIPTION
**Pull request by stefaniapedrazzi**
Fix #582: it seems that using `QDoubleSpinBox::text().toDouble()` instead of `QDoubleSpinBox::value()` solves the issue.

### CODE REVIEW

**Comment by stefaniapedrazzi** [comment]
:+1:

**Comment by omichel** [comment]
:eyes:

File: src/webots/scene_tree/WbDoubleEditor.cpp [hunk 1]
```diff
@@ -69,8 +69,9 @@ void WbDoubleEditor::resetFocus() {
 }
 
 void WbDoubleEditor::applyIfNeeded() {
-  if (field() && ((field()->hasRestrictedValues() && mDouble != mComboBox->currentText().toDouble()) ||
-                  (!field()->hasRestrictedValues() && mDouble != mSpinBox->value())))
+  if (field() &&
+      ((field()->hasRestrictedValues() && mDouble != mComboBox->currentText().toDouble()) ||
+       (!field()->hasRestrictedValues() && fabs(mDouble - mSpinBox->value()) > WbPrecision::epsilon(WbPrecision::DOUBLE_MAX))))
```
Comment (omichel) [inline src/webots/scene_tree/WbDoubleEditor.cpp]:
```suggestion
       (!field()->hasRestrictedValues() && mDouble != mSpinBox->text().toDouble())))
```
Isn't it simpler and more consistent this way?

**Comment by stefaniapedrazzi** [comment]
:+1:

**Pull request merged**